java/Anton-Kozyk/src/com/zoolatech/lecture2/tasks/_2/Task2.java

```diff
@@ -0,0 +1,190 @@
+                    }
+                }
+                case E -> {
+                    if (x + step < Width) {
+                        x += step;
+                    } else {
+                        x = Width - 1;
+                        System.out.println("You can't go East anymore, please change direction.");
+                    }
+                }
+                case N -> {
+                    if (y - step >= 0) {
+                        y -= step;
+                    } else {
+                        y = 0;
+                        System.out.println("You can't go North anymore, please change direction.");
+                    }
+                }
+                case W -> {
+                    if (x - step >= 0) {
+                        x -= step;
+                    } else {
+                        x = 0;
+                        System.out.println("You can't go West anymore, please change direction.");
+                    }
+                }
+            }
+            getPosition();
+        } else {
+            System.out.println("Nothing happened...");
+        }
+    }
+
+    public void turnLeft() {
+        direction = switch (direction) {
+            case S -> E;
+            case E -> N;
+            case N -> W;
+            case W -> S;
+            default -> throw new IllegalStateException("Unknown direction: " + direction);
+        };
+        System.out.println("Direction changed to " + direction + "\n");
+    }
+
+    public void turnRight() {
+        direction = switch (direction) {
+            case S -> W;
+            case W -> N;
+            case N -> E;
+            case E -> S;
+            default -> throw new IllegalStateException("Unknown direction: " + direction + "\n");
+        };
+        System.out.println("Direction changed to " + direction);
+    }
+
+    public void printPosition() {
+        System.out.println(
+                        """
+                        Compass
+                           N
+                        W     E
+                           S""");
+
+        getPosition();
+        System.out.println("Direction: " + direction + "\n");
+        for (int i = 0; i < Height; i++) {
+            for (int j = 0; j < Width; j++) {
+                if (i == y && j == x) {
+                    System.out.print(switch (direction) {
+                        case S -> "[\\/ ]";
+                        case W -> "[ < ]";
+                        case N -> "[/\\ ]";
+                        case E -> "[ > ]";
+                        default -> throw new IllegalStateException("Unexpected direction: " + direction);
+                    });
+                } else {
+                    System.out.print("[" + i + ":" + j + "]");
+                }
+            }
+            System.out.println();
+        }
+        System.out.println();
+    }
+
+    class RobotsPosition {
+        int x = Robot.this.x;
```

Comment:
Nice idea to create a separate class, but it's better to create a constructor and pass these coordinates to it. Currently, classes are too tightly coupled.